engine/gamesys/src/gamesys/scripts/script_model.cpp

```diff
@@ -610,18 +610,64 @@ namespace dmGameSystem
         return 1;
     }
 
+    /*# get the AABB of the meshes
+     * Get AABB of the meshes.
+     * AABB information return as a table with `min` and `max` fields, where `min` and `max` has type `vmath.vector4`.
+     *
+     * @name model.get_aabb
+     * @param url [type:string|hash|url] the model
+     * @return aabb [type:table] A table containing info about all AABB in the format <hash(mesh_id), aabb_info>
+     * @examples
+     *
+     * ```lua
+     * model.get_aabb("#model") -> { hash("Sword") = { min = vmath.vector3(-0.5, -0.5, 0), max = vmath.vector3(0.5, 0.5, 0) }, hash("Shield") = { min = vmath.vector3(-0.5, -0.5, -0.5), max = vmath.vector3(0.5, 0.5, 0.5) } }
+     * ```
+     */
+    static int LuaModelComp_GetAabb(lua_State* L)
+    {
+        DM_LUA_STACK_CHECK(L, 1);
+        ModelComponent* component = 0;
+        dmGameObject::HInstance sender_instance = CheckGoInstance(L);
+        dmGameObject::HCollection collection = dmGameObject::GetCollection(sender_instance);
+        dmGameObject::GetComponentFromLua(L, 1, collection, MODEL_EXT, (dmGameObject::HComponent*)&component, 0, 0);
+        if (!component)
+        {
+            return luaL_error(L, "the component '%s' could not be found", lua_tostring(L, 1));
+        }
+        lua_newtable(L);
```

Comment:
If you know the count (line below), you can create the table with `lua_createtable(L, 0, n)`
https://www.lua.org/manual/5.1/manual.html#lua_createtable